### ISSUE TITLE
New functions and views in Public are not removed by the uninstaller (WM)

### ISSUE THREAD

**Issue by srrollo** [DASSL/ClassDB#146]
[`addFrequentViews.sql`](https://github.com/DASSL/ClassDB/blob/add-frequent-views/src/addFrequentViews.sql) adds several new objects to `Public` that are not removed by `removeFromDB.sql`. I suggest that we simply add `DROP * IF EXISTS` statements to `removeFromDB.sql` for each object needing to be dropped. An initial list of such objects is:
- [ ]  `Public.describe(VARCHAR(63))`
- [ ] `Public.getMyActivitySummary()`
- [ ] `Public.getMyDDLActivity()`
- [ ] `Public.getMyConnectionActivity()`
- [ ] `Public.getMyActivity()`
- [ ] `Public.MyDDLActivity`
- [ ] `Public.MyConnectionActivity`
- [ ] `Public.MyActivity`
- [ ] `Public.MyActivitySummary`

**Comment by smurthys** [comment]
I gathered some information about generating a script to list/drop functions. The internal function `oidvectortypes` is the magic sauce as seen in this query:

```sql
SELECT 'DROP FUNCTION ' || ns.nspname || '.' || p.proname 
       || '(' || oidvectortypes(p.proargtypes) || ');'
FROM pg_proc p JOIN pg_namespace ns ON p.pronamespace = ns.oid
WHERE ns.nspname = 'classdb'
ORDER BY p.proname;
```

Useful sources;
- https://stackoverflow.com/questions/10591113/drop-all-functions-from-postgres-database#10591166
- http://www.postgresonline.com/journal/archives/74-How-to-delete-many-functions.html
- https://www.postgresql.org/message-id/Pine.LNX.4.21.0007072058000.587-100000%40localhost.localdomain

**Comment by srrollo** [comment]
This is an awesome function @smurthys. I was also thinking that somehow listing all functions owned by `ClassDB` would be the best way to solve this issue.

**Comment by srrollo** [comment]
After investigating #147, it seems these two issues are linked fairly tightly. I will post an update here later.

**Comment by srrollo** [comment]
I did some experimenting after our discussion, it seems one fairly simple solution we didn't discuss is to use `DROP OWNED BY ClassDB`. If this encounters any dependency issues, it throws a very helpful exception, for example:

`DETAIL:  view m2ins01.insview depends on table classdb.rolebase`

The one exception to this is that objects that are owned by superusers, for example, the event triggers and log import function, must be explicitly dropped first. I've pushed an implementation of this to the [`remove-uninstaller-uname-dep`](https://github.com/DASSL/ClassDB/blob/remove-uninstaller-uname-dep/src/removeFromDB.sql) branch. It seems to work as intended the testing I've performed locally.
